spec/iso8601/time_interval_spec.rb

```diff
@@ -0,0 +1,366 @@
+      @big = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT2H')
+    end
+
+    it "should raise TypeError when compared object is not a ISO8601::TimeInterval or Numeric" do
+      expect { @small < 'Hello!' }.to raise_error(ArgumentError)
+      expect { @small > 'Hello!' }.to raise_error(ArgumentError)
+    end
+
+    it "should check what interval is bigger" do
+      expect(@small <=> @big).to eq(-1)
+      expect(@big <=> @small).to eq(1)
+      expect(@big <=> @big).to eq(0)
+
+      expect(@small > @big).to be_falsy
+      expect(@big > @small).to be_truthy
+      expect(@small > @small).to be_falsy
+    end
+
+    it "should check if interval is bigger or equal than other" do
+      expect(@small >= @big).to be_falsy
+      expect(@big >= @small).to be_truthy
+      expect(@small >= @small).to be_truthy
+    end
+
+    it "should check what interval is smaller" do
+      expect(@small < @big).to be_truthy
+      expect(@big < @small).to be_falsy
+      expect(@small < @small).to be_falsy
+    end
+
+    it "should check if interval is smaller or equal than other" do
+      expect(@small <= @big).to be_truthy
+      expect(@big <= @small).to be_falsy
+      expect(@small <= @small).to be_truthy
+    end
+
+    it "should check if the intervals are equals" do
+      expect(@small == @small).to be_truthy
+      expect(@small == @small.to_f).to be_falsy
+      expect(@small == @big).to be_falsy
+      expect(@small == @big.to_f).to be_falsy
+    end
+  end
+
+  describe "#eql?" do
+    it "should be equal only when start_time and end_time are the same" do
+      interval = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT1H')
+      interval2 = ISO8601::TimeInterval.new('2007-03-01T14:00:00Z/PT1H')
+      interval3 = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT1H')
+
+      expect(interval.eql?(interval2)).to be_falsy
+      expect(interval.eql?(interval3)).to be_truthy
+    end
+  end
+
+  describe "#include?" do
+    it "raise TypeError when the parameter is not valid" do
+      ti = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/PT1H')
+      expect { ti.include?('hola') }.to raise_error(ISO8601::Errors::TypeError)
+      expect { ti.include?(123) }.to raise_error(ISO8601::Errors::TypeError)
+    end
+
+    it "should check if a DateTime is included" do
+      ti = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/P1DT1H')
+      included = DateTime.new(2007, 3, 1, 15, 0, 0)
+      included_iso8601 = ISO8601::DateTime.new('2007-03-01T18:00:00Z')
+      not_included = DateTime.new(2007, 2, 1, 15, 0, 0)
+      not_included_iso8601 = ISO8601::DateTime.new('2007-03-01T11:00:00Z')
+
+      expect(ti.include?(included)).to be_truthy
+      expect(ti.include?(included_iso8601)).to be_truthy
+      expect(ti.include?(not_included)).to be_falsy
+      expect(ti.include?(not_included_iso8601)).to be_falsy
+    end
+
+    it "should check if an interval is included" do
+      ti = ISO8601::TimeInterval.new('2007-03-01T13:00:00Z/P1DT1H')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.